shared/src/main/scala/pl/metastack/metarx.scala

```diff
@@ -46,4 +52,59 @@ package object metarx extends OptImplicits {
     def &&(rch: ReadChannel[Boolean]): ReadChannel[Boolean] = rch && boolVal
     def ||(rch: ReadChannel[Boolean]): ReadChannel[Boolean] = rch || boolVal
   }
+
+  implicit class ReadChannelNumericExtensions[T: Numeric](rch: ReadChannel[T])(implicit num: Numeric[T]) extends ReadChannelExtensionBase(rch) {
+    import num._
+
+    def +(other: ReadChannel[T]): ReadChannel[T] = otherWithOperation(other, plus)
+    def +(argument: T): ReadChannel[T] = TWithOperation(argument, plus)
+    def -(other: ReadChannel[T]): ReadChannel[T] = otherWithOperation(other, minus)
+    def -(argument: T): ReadChannel[T] = TWithOperation(argument, minus)
+    def *(other: ReadChannel[T]): ReadChannel[T] = otherWithOperation(other, times)
+    def *(argument: T): ReadChannel[T] = TWithOperation(argument, times)
+
+    def unary_-(other: ReadChannel[T]): ReadChannel[T] = rch.map(-_)
+    def abs(): ReadChannel[T] = rch.map(num.abs)
+
+    def toInt: ReadChannel[Int] = rch.map(_.toInt)
+    def toLong: ReadChannel[Long] = rch.map(_.toLong)
+    def toFloat: ReadChannel[Float] = rch.map(_.toFloat)
+    def toDouble: ReadChannel[Double] = rch.map(_.toDouble)
+  }
+
+  implicit class NumericExtensions[T: Numeric](value: T)(implicit num: Numeric[T]) {
+    import num._
+
+    def +(other: ReadChannel[T]): ReadChannel[T] = other.map(plus(value, _))
+    def -(other: ReadChannel[T]): ReadChannel[T] = other.map(minus(value, _))
+    def *(other: ReadChannel[T]): ReadChannel[T] = other.map(times(value, _))
```

Comment:
There is no division in Numeric. It is added in child classes Integral and Fractional. See later extensions.